deepr/layers/nce_loss.py

```diff
@@ -59,19 +59,18 @@ def forward(self, tensors, mode: str = None):
             Negative Sampling  loss
         """
         positives, negatives, mask, weights = tensors
-        mask = tf.cast(mask, tf.float32)
         true_losses = tf.nn.sigmoid_cross_entropy_with_logits(labels=tf.ones_like(positives), logits=positives)
         sampled_losses = tf.nn.sigmoid_cross_entropy_with_logits(labels=tf.zeros_like(negatives), logits=negatives)
         # filter the values that correspond to mask
-        negative_number = tf.reduce_sum(mask, -1)
-        sampled_losses = tf.multiply(sampled_losses, mask)
+        negative_number = tf.reduce_sum(tf.to_float(mask), -1)
+        sampled_losses = tf.multiply(sampled_losses, tf.to_float(mask))
```

Comment:
well done spotting the same thing here! not sure about any impact on speed, but maybe doing only one `mask_float = ...`  would be better